sklearn/base.py

```diff
@@ -57,14 +57,22 @@ def clone(estimator, safe=True):
     # XXX: not handling dictionaries
     if estimator_type in (list, tuple, set, frozenset):
         return estimator_type([clone(e, safe=safe) for e in estimator])
-    elif not hasattr(estimator, 'get_params') or isinstance(estimator, type):
+    elif not hasattr(estimator, 'get_params'):
         if not safe:
             return copy.deepcopy(estimator)
         else:
             raise TypeError("Cannot clone object '%s' (type %s): "
```

Comment:
What I add in mind was to make an additional check here

```suggestion
            if issubclass(estimator, BaseEstimator):
                raise TypeError("You should provide an instance of scikit-learn estimator instead of a class.")
            else:
                raise TypeError("previous error message")
            raise TypeError("Cannot clone object '%s' (type %s): "
```

Reply:
Thanks I tried following this